### PR TITLE
Make verify-codegen required

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

successful job here: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/5137/pull-test-infra-verify-codegen/30/?log#log
/assign @cjwagner @fejta

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@cjwagner @fejta

**Comment by stevekuznetsov** [comment]
(or @BenTheElder )
fixed this last night

**Comment by fejta** [comment]
Please make this run all the time but skip reporting, this doesn't yet look safe to require on all PRs: https://k8s-testgrid.appspot.com/presubmits-test-infra#verify-codegen

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 1]
```diff
@@ -205,7 +205,6 @@ presubmits:
     trigger: "/test pull-test-infra-verify-codegen"
     run_if_changed: '(prow/(apis|client)|vendor)/.*$'
```
Comment (fejta) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: Also if we believe we have it working now let's just run it on every PR.

I shouldn't have to modify this regex every time I add a new controller somewhere in the codebase.

**Comment by fejta** [comment]
Actually this is looking good enough to require: https://k8s-testgrid.appspot.com/presubmits-test-infra#verify-codegen

Can we drop the run_if_changed though and run it on all PRs?

**Comment by stevekuznetsov** [comment]
Seems like a waste of resources to run all the time since it can only possibly fail if you change files in those directories but if you don't mind that then yes I can `run_always: true` -- @fejta WDYT?

**Comment by fejta** [comment]
I'm more interested in consistency than optimization, especially for a job that completes in 20s.

We don't know what future commit is going to invalidate the statement that only `(prow/(apis|client)|vendor)/.*$'` can cause code to be regenerated. And the person who writes that PR to find this out by this check failing on the PR. This is better than some random other person finding out at some unknown later time.

**Comment by stevekuznetsov** [comment]
Makes sense. @fejta changed to `run_always: true`

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 2]
```diff
@@ -203,9 +203,8 @@ presubmits:
     - master
     rerun_command: "/test pull-test-infra-verify-codegen"
     trigger: "/test pull-test-infra-verify-codegen"
-    run_if_changed: '(prow/(apis|client)|vendor)/.*$'
+    run_always: true
```
Comment (cjwagner) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: `always_run` :smile:

Comment (stevekuznetsov) [reply config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: yuuuuuuck

**Comment by stevekuznetsov** [comment]
@cjwagner fixed :)

**Pull request merged**